sdks/python/apache_beam/transforms/util.py

```diff
@@ -949,6 +952,97 @@ def is_compat_version_prior_to(options, breaking_change_version):
   return False
 
 
+def _default_window_reify_functions(should_keep_paneinfo):
+  globally_windowed = window.GlobalWindows.windowed_value(None)
+  MIN_TIMESTAMP = window.MIN_TIMESTAMP
+
+  if should_keep_paneinfo:
+
+    def reify_metadata(
```

Comment:
Nothing about this needs to be defined in a nested context. For readability, just define the functions at the top level and then the if statement can be concise and clear.

Reply:
Ah, yes, I can explain: The update compatibility breaks when the encoded state that is shuffled changes:

- The running pipeline will have a bunch of records that just include the timestamp and window.
- The pipeline we are trying to replace it with will be expecting records that include the fuller metadata including pane info.

So the idea of the new flag is that whenever we want to make a change that is incompatible, we can fork the existing code and put it in the attic, and at the start of `expand` we can redirect control flow to that code if a user requests it.